File: CRUDdemo/app/src/main/java/com/example/crud_demo/MainActivity.java
package com.example.crud_demo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;

import com.example.crud_demo.helper.DatabaseHelper;
import com.example.crud_demo.model.BoardGame;

public class MainActivity extends AppCompatActivity {

    DatabaseHelper databaseHelper;
    private Button crt_btn;
    private ListView listViewOfGame;
    private TextView calculateRevenue;

    ArrayAdapter gameArrayAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        databaseHelper = new DatabaseHelper(this);
        crt_btn= findViewById(R.id.createProduct);
        listViewOfGame=findViewById(R.id.listViewGames);
        calculateRevenue=findViewById(R.id.revenueNum);

        calculateRevenue.setText(String.valueOf("MISSING"));

        gameArrayAdapter = new ArrayAdapter<BoardGame>(MainActivity.this,
                android.R.layout.simple_list_item_1, databaseHelper.getAllGame());
        listViewOfGame.setAdapter(gameArrayAdapter);

        // Action for create button
        crt_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               // openCreateProductPage();
            }
        });
    }



}

File: FileDemo/app/src/main/java/com/example/filedemo/MainActivity.kt
package com.example.filedemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import java.lang.Exception

class MainActivity : AppCompatActivity() {

    lateinit var btnSave:Button
    lateinit var btnLoad:Button
    lateinit var tv:TextView
    lateinit var et:EditText

    private var data:String? = null
    private val file = "myData"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnSave = findViewById(R.id.btnSave)
        btnLoad = findViewById(R.id.btnLoad)

        et = findViewById(R.id.etInput)
        tv = findViewById(R.id.tvOutput)

        btnSave.setOnClickListener(View.OnClickListener {
            data = et.text.toString()
            try {
                val fOut = openFileOutput(file, MODE_PRIVATE)
                fOut.write(data!!.toByteArray())
                fOut.close()
                Toast.makeText(baseContext, "File saved!", Toast.LENGTH_LONG).show()
            } catch (e: Exception) {
                // do something else...
                e.printStackTrace()
            }
        })

        btnLoad.setOnClickListener(View.OnClickListener {
            try {
                val fIn = openFileInput(file)
                var c: Int
                var temp = ""

                while (fIn.read().also { c = it } != -1) {
                    temp += Character.toString(c.toChar())
                }

                tv.setText(temp)
                Toast.makeText(baseContext, "File loaded!", Toast.LENGTH_LONG).show()

            } catch (e: Exception) {
                // do something else...
                e.printStackTrace()
            }
        })
    }
}